src/main/java/me/arminb/sara/dao/UserDAO.java

```diff
@@ -0,0 +1,16 @@
+package me.arminb.sara.dao;
+
+import me.arminb.sara.entities.User;
+import org.bson.types.ObjectId;
+
+import java.util.List;
+
+public interface UserDAO {
+    public List<User> findAll() throws DataAccessException;
+    public User findById(ObjectId id) throws DataAccessException;
+    public List<User> find(ObjectId id, String username, String email) throws DataAccessException;
+    public User create(User user) throws DataAccessException;
+    public boolean delete(ObjectId id) throws DataAccessException;
+    public User update(User user) throws DataAccessException;
+
```

Comment:
It is possible to merge create and update methods into
>> public User save(User user) throws DataAccessException;
method. This method can check if the id field is empty or not and then issue an update or create accordingly. This will centralize the modification logic in one place.